app/code/Magento/CatalogInventory/Model/StockStateProvider.php

```diff
@@ -80,6 +80,9 @@ public function verifyStock(StockItemInterface $stockItem)
         ) {
             return false;
         }
+        if ($stockItem->getQty() > 0 && !$stockItem->getIsInStock()) {
+            return false;
```

Comment:
Why this logic needed in `CatalogInventory` if only `ImportExport` is affected?

Reply:
That specific function will have a levels of the stock check while product import/export and which does not cause any other problem for other behaviors